### ISSUE TITLE
Replace old collections

### ISSUE THREAD

**Issue by mcavigelli** [MarcosMeli/FileHelpers#235]
In order to get more .NET Standard 1.6 compatible (see #213), we would like to replace old collections not present in .NET Standard 1.6:

- Hashtable (non generic)
- ArrayList
- StringCollection

These old collections must be replaced with generic collections that are present in .NET Standard 1.6 and .NET Framework, if possible 2.0.

These changes must be incorporated into the master branch.

**Comment by mcavigelli** [comment]
Hi Guillaume

Thank you for your work.
I have noticed that you assigned yourself to three issues. I think it would be better to just pick one issue at a time. This alleviates tracking progress and collaboration in the project.

In my opinion an issue should be solved, a pull request sent and then integrated into the master branch.
The work in progress should be kept small.

Working on one issue at a time and then sending / reviewing the change will allow better and more immediate feedback. The assigned issues are blocked and other people cannot work on it. That also
means that people cannot work in parallel on different issues.

Merci beaucoup, Matthias

**Comment by guillaumejay** [comment]
Hello,

As you want, especially since I was not able to start work as them as I intended.

I'll pick one now.

Cordialement,
Guillaume

**Comment by guillaumejay** [comment]
I'm looking at this one, do you agree with :
HashTable => Dictionary<> ([but it's going to be in 2.0](https://apisof.net/catalog/System.Collections.Hashtable)   )
ArrayList => List<> ([but it's going to be in 2.0 ]( https://apisof.net/catalog/System.Collections.ArrayList) )
StringCollection => List<String> ( [but  ...])(https://apisof.net/catalog/System.Collections.Specialized.StringCollection)

Eventually everything will be in Standard 2.0, but should we wait for it ?
(Main breaking point right now is that Hashtable[KeyNotPresent]== null, whereas Dictionary[KeyNotPresent] throws an exception : thanks for unit tests ! )

**Comment by mcavigelli** [comment]
Salut Guillaume

I agree with your proposed mapping.

 We should get rid of these .NET 1.0 collections anyway:
- HashTable and ArrayList require a lot of casting as they both deal with _object_. 
- StringCollection does not implement IEnumerable<string> (I am not sure anymore, if we
  are really using them).

Amicalement et merci, Matthias

**Comment by mcavigelli** [comment]
Hâlo Guillaume

As it turned out we are targeting .NET Core 2.0.
Most of the aforementioned or required collections are contained in .NET Core 2.0.
That means that this is issue has very low priority or little relevance.

Do you mind, if I close this issue?

Matthias

**Comment by guillaumejay** [comment]
Of course I don't mind, especially since I never finished it (shame on me)
It was irrelevant anyway with 2.0
